File: src/types/sftp.ts
import { z } from 'zod';

export const sftpEntrySchema = z.object({
  name: z.string(),
  path: z.string(),
  isDirectory: z.boolean(),
  size: z.number(),
  modifiedAt: z.string().optional(),
});

export type SftpEntry = z.infer<typeof sftpEntrySchema>;

export const listDirResponseSchema = z.object({
  entries: z.array(sftpEntrySchema),
  resolvedPath: z.string(),
});

export type ListDirResponse = z.infer<typeof listDirResponseSchema>;

export const recursiveFileEntrySchema = z.object({
  path: z.string(),
  name: z.string(),
  relativePath: z.string(),
  size: z.number(),
  modifiedAt: z.string().optional(),
});

export type RecursiveFileEntry = z.infer<typeof recursiveFileEntrySchema>;

File: src-tauri/src/models/sftp.rs
use serde::Serialize;

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct SftpEntry {
    pub name: String,
    pub path: String,
    pub is_directory: bool,
    pub size: u64,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub modified_at: Option<String>,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ListDirResponse {
    pub entries: Vec<SftpEntry>,
    pub resolved_path: String,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct RecursiveFileEntry {
    pub path: String,
    pub name: String,
    pub relative_path: String,
    pub size: u64,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub modified_at: Option<String>,
}
